src/sbd-common.c

```diff
@@ -173,20 +231,27 @@ watchdog_tickle(void)
 static int
 watchdog_init_fd(char *wddev, int timeout)
 {
-	int wdfd;
+    int wdfd;
+
+    wdfd = open(wddev, O_WRONLY);
+    if (wdfd >= 0) {
+        if (((timeout >= 0) && (watchdog_init_interval_fd(wdfd, timeout) < 0)) ||
+            (watchdog_tickle_fd(wdfd, wddev) < 0)) {
+            close(wdfd);
+            return -1;
+        }
+    } else {
+        struct stat statbuf;
 
-	wdfd = open(wddev, O_WRONLY);
-	if (wdfd >= 0) {
-		if (((timeout >= 0) && (watchdog_init_interval_fd(wdfd, timeout) < 0))
-					|| (watchdog_tickle_fd(wdfd, wddev) < 0)) {
-			close(wdfd);
-			return -1;
-		}
-	} else {
-		cl_perror("Cannot open watchdog device '%s'", wddev);
-		return -1;
-	}
-	return wdfd;
+        if(!stat(wddev, &statbuf) && S_ISCHR(statbuf.st_mode) &&
```

Comment:
Presumably this stat() might also trigger SELinux denials, but maybe that's acceptable here, and likely under /dev anyway

Reply:
Well the critical stats are those done when traversing over /proc or /sys with links to arbitrary places in the filesystem.